pytensor/link/jax/dispatch/slinalg.py

```diff
@@ -54,7 +54,21 @@ def solve(a, b):
             dl = jax.numpy.diagonal(a, offset=-1, axis1=-2, axis2=-1)
             d = jax.numpy.diagonal(a, offset=0, axis1=-2, axis2=-1)
             du = jax.numpy.diagonal(a, offset=1, axis1=-2, axis2=-1)
-            return jax.lax.linalg.tridiagonal_solve(dl, d, du, b, lower=lower)
+            # jax requires dl and du to have the same shape as d
+            dl = jax.numpy.pad(dl, (1, 0))
+            du = jax.numpy.pad(du, (0, 1))
```

Comment:
In #1416 I found that pad is really slow in numpy. Not sure about jax, but it might be better to try some other ways to add this zero (concat maybe?)

Reply:
The advantage of compiled code is that it can be reasoned about once with slow python instead of in every call

Reply:
So we're keeping this the way it is then?

Reply:
yes, unless someone benchmarks and finds it is slow, which I wouldn't bet at first